src/traces/pie/base_plot.js

```diff
@@ -16,8 +16,7 @@ exports.name = 'pie';
 exports.plot = function(gd) {
     var Pie = Registry.getModule('pie');
     var cdPie = getModuleCalcData(gd.calcdata, Pie)[0];
-
-    if(cdPie.length) Pie.plot(gd, cdPie);
+    Pie.plot(gd, cdPie);
```

Comment:
@etpinard Could you please elaborate why the condition check is no longer required here?

Reply:
Because we need `Pie.plot` to `.exit().remove()` the old pie traces.